api/src/main/java/ai/djl/ndarray/index/NDIndex.java

```diff
@@ -373,7 +373,7 @@ private int addIndexItem(String indexItem, int argIndex, Object[] args) {
                     if (array.getDataType() == DataType.BOOLEAN) {
                         indices.add(new NDIndexBooleans(array));
                         return argIndex + 1;
-                    } else if (array.getDataType().isInteger()) {
+                    } else {
```

Comment:
Would this cause issues if DataType is UNKNOWN or STRING? https://github.com/deepjavalibrary/djl/blob/master/api/src/main/java/ai/djl/ndarray/types/DataType.java#L28-L34 seems like what we need to cover in this code, and not UNKNOWN or STRING

Reply:
Thanks for spotting this. I'll update it.